Source/WebCore/crypto/gcrypt/CryptoAlgorithmX25519GCrypt.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2023 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2,1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "CryptoAlgorithmX25519.h"
+
+#if ENABLE(WEB_CRYPTO)
+
+#include "CryptoKeyOKP.h"
+#include "GCryptRFC7748.h"
+#include <pal/crypto/gcrypt/Handle.h>
+#include <pal/crypto/gcrypt/Utilities.h>
+
+namespace WebCore {
+
+static std::optional<Vector<uint8_t>> gcryptDerive(const Vector<uint8_t>& baseKey, const Vector<uint8_t>& publicKey, size_t)
```

Comment:
This can be simplified to not receive an unused size parameter and also simply return the `X25519` call.

Reply:
I''ll do this refactoring in a follow up patch, thanks for the suggestion.

Reply:
Finally done in this PR